### PR TITLE
feat: Mileage/ Per Diem expenses to depend on category

### PR DESCRIPTION
**Pull request by arjunaj5**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=e_36Onw)

DOC section: https://www.notion.so/fyleuniverse/Application-Engineering-2b25fffe7b3641c3a5d90d4417a0e360?pvs=4#bb4e8bc013f140999e0937e99149c29d

https://app.clickup.com/t/85ztzt5y6
### <samp>🤖 Generated by Copilot at 1b53cfa</samp>

This pull request enables the creation of mileage and per diem expenses using special categories from the platform. It modifies the `dashboard.page.ts` and the `categories.service.ts` files to fetch and use these categories.

### <samp>🤖 Generated by Copilot at 1b53cfa</samp>

> _Sing, O Muse, of the skillful coder who devised_
> _A clever method `getMileageOrPerDiemCategories`_
> _To fetch the platform categories that are enabled_
> _And have the system category of Mileage or Per Diem._

### <samp>🤖 Generated by Copilot at 1b53cfa</samp>

*  Add a new method `getMileageOrPerDiemCategories` to fetch the special categories for mileage and per diem expenses ([link](https://github.com/fylein/fyle-mobile-app/pull/2461/files?diff=unified&w=0#diff-00ea7bf26983400bc3a2227bb4c1695132c71429b10ebea74274d01ddf41e6eaR81-R92))
*  Import `forkJoin` operator, `PlatformCategory` model, and `CategoriesService` in `dashboard.page.ts` file ([link](https://github.com/fylein/fyle-mobile-app/pull/2461/files?diff=unified&w=0#diff-6988ea80c678c948087ba59efeaaa42bdc2f4e34c61952e9297aa625645a4806L2-R2), [link](https://github.com/fylein/fyle-mobile-app/pull/2461/files?diff=unified&w=0#diff-6988ea80c678c948087ba59efeaaa42bdc2f4e34c61952e9297aa625645a4806R22-R23))
*  Declare a new property `specialCategories$` to store the special categories observable ([link](https://github.com/fylein/fyle-mobile-app/pull/2461/files?diff=unified&w=0#diff-6988ea80c678c948087ba59efeaaa42bdc2f4e34c61952e9297aa625645a4806R46-R47))
*  Inject `CategoriesService` as a dependency in the constructor of `DashboardPage` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2461/files?diff=unified&w=0#diff-6988ea80c678c948087ba59efeaaa42bdc2f4e34c61952e9297aa625645a4806R73))
*  Combine `orgSettings$` and `specialCategories$` observables using `forkJoin` and pass `allowedExpenseTypes` object to `setupActionSheet` method ([link](https://github.com/fylein/fyle-mobile-app/pull/2461/files?diff=unified&w=0#diff-6988ea80c678c948087ba59efeaaa42bdc2f4e34c61952e9297aa625645a4806L119-R135))
*  Modify `setupActionSheet` method to check `allowedExpenseTypes` object and display action sheet buttons accordingly ([link](https://github.com/fylein/fyle-mobile-app/pull/2461/files?diff=unified&w=0#diff-6988ea80c678c948087ba59efeaaa42bdc2f4e34c61952e9297aa625645a4806L232-R248))

## Clickup
Please add link here

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/services/categories.service.ts [hunk 1]
```diff
@@ -78,6 +78,19 @@ export class CategoriesService {
     );
   }
 
+  @Cacheable()
+  getMileageOrPerDiemCategories(): Observable<PlatformCategory[]> {
+    const data = {
+      params: {
+        is_enabled: 'eq.' + true,
```
Comment (mvaishnavi) [inline src/app/core/services/categories.service.ts:85]: `is_enabled: 'eq.true'`

File: src/app/fyle/dashboard/dashboard.service.spec.ts [hunk 2]
```diff
@@ -29,6 +29,7 @@ import { expectedAssignedCCCStats } from '../../core/mock-data/ccc-expense.detai
 import { of } from 'rxjs';
```
Comment (arjunaj5) [inline src/app/fyle/dashboard/dashboard.service.spec.ts:29]: will remove these unwanted changes

File: src/app/fyle/my-expenses/my-expenses.page.spec.ts [hunk 3]
```diff
@@ -104,8 +104,10 @@ import { FyDeleteDialogComponent } from 'src/app/shared/components/fy-delete-dia
 import { getElementRef } from 'src/app/core/dom-helpers';
 import { transactionDatum1, transactionDatum3 } from 'src/app/core/mock-data/stats-response.data';
 import { uniqueCardsParam } from 'src/app/core/mock-data/unique-cards.data';
+import { allowedExpenseTypes } from 'src/app/core/mock-data/allowed-expense-types.data';
+import { CategoriesService } from 'src/app/core/services/categories.service';
 
-describe('MyExpensesPage', () => {
+fdescribe('MyExpensesPage', () => {
```
Comment (arjunaj5) [inline src/app/fyle/my-expenses/my-expenses.page.spec.ts:110]: will remove this

File: src/app/fyle/dashboard/dashboard.page.scss [hunk 4]
```diff
@@ -60,5 +60,12 @@
   &--action-shortcut {
     padding-right: 12px;
     padding-top: 2px;
+
+    &--skeleton-icon {
+      margin-right: 12px;
+      width: 24px;
+      height: 24px;
+      border-radius: 5px;
```
Comment (Dimple16) [inline src/app/fyle/dashboard/dashboard.page.scss:68]: should it be 4/6?

**Review by Dimple16 — APPROVED** [review]
minor comment

**Comment by Dimple16** [comment]
@arjunaj5 please check if the tests are failing because of your changes?

**Review by mvaishnavi — APPROVED** [review]

**Pull request merged**